@codex 下記２種類のCIエラーを解消させてください。

---

0s
8s
Run pytest -q --no-cov tests/test_security_headers.py

==================================== ERRORS ====================================
_______________ ERROR collecting tests/test_security_headers.py ________________
tests/test_security_headers.py:83: in <module>
    from backend.main import create_app
apps/backend/backend/main.py:386: in <module>
    app = create_app()
          ^^^^^^^^^^^^
apps/backend/backend/main.py:380: in create_app
    app.add_event_handler("shutdown", _on_shutdown)
    ^^^^^^^^^^^^^^^^^^^^^
E   AttributeError: 'FastAPI' object has no attribute 'add_event_handler'. Did you mean: 'add_exception_handler'?
------------------------------- Captured stderr --------------------------------
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /home/runner/nltk_data...
[nltk_data]   Unzipping taggers/averaged_perceptron_tagger.zip.
[nltk_data] Downloading package cmudict to /home/runner/nltk_data...
[nltk_data]   Unzipping corpora/cmudict.zip.
{"event": "ALLOWED_HOSTS init raw='api.ci.wordpack.example.com,localhost,testserver' parsed=['api.ci.wordpack.example.com', 'localhost', 'testserver']", "timestamp": "2026-04-05T15:10:39.870621Z", "level": "info", "severity": "INFO"}
{"reason": "config_flag", "event": "session_auth_disabled", "timestamp": "2026-04-05T15:10:39.871134Z", "level": "warning", "severity": "WARNING"}
=========================== short test summary info ============================
ERROR tests/test_security_headers.py - AttributeError: 'FastAPI' object has no attribute 'add_event_handler'. Did you mean: 'add_exception_handler'?
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
Error: Process completed with exit code 2.

---

8s
Run npx playwright test \
[WebServer] [nltk_data] Downloading package averaged_perceptron_tagger to
[WebServer] [nltk_data]     /home/runner/nltk_data...
[WebServer] [nltk_data]   Unzipping taggers/averaged_perceptron_tagger.zip.
[WebServer] [nltk_data] Downloading package cmudict to /home/runner/nltk_data...
[WebServer] [nltk_data]   Unzipping corpora/cmudict.zip.
[WebServer] {"event": "ALLOWED_HOSTS init raw='*' parsed=['*']", "timestamp": "2026-04-05T15:55:29.735873Z", "level": "info", "severity": "INFO"}
[WebServer] Traceback (most recent call last):
[WebServer]   File "<frozen runpy>", line 198, in _run_module_as_main
[WebServer]   File "<frozen runpy>", line 88, in _run_code
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/uvicorn/__main__.py", line 4, in <module>
[WebServer]     uvicorn.main()
[WebServer]     ~~~~~~~~~~~~^^
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/click/core.py", line 1485, in __call__
[WebServer]     return self.main(*args, **kwargs)
[WebServer]            ~~~~~~~~~^^^^^^^^^^^^^^^^^
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/click/core.py", line 1406, in main
[WebServer]     rv = self.invoke(ctx)
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/click/core.py", line 1269, in invoke
[WebServer]     return ctx.invoke(self.callback, **ctx.params)
[WebServer]            ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/click/core.py", line 824, in invoke
[WebServer]     return callback(*args, **kwargs)
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/uvicorn/main.py", line 433, in main
[WebServer]     run(
[WebServer]     ~~~^
[WebServer]         app,
[WebServer]         ^^^^
[WebServer]     ...<47 lines>...
[WebServer]         h11_max_incomplete_event_size=h11_max_incomplete_event_size,
[WebServer]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[WebServer]     )
[WebServer]     ^
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/uvicorn/main.py", line 606, in run
[WebServer]     server.run()
[WebServer]     ~~~~~~~~~~^^
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/uvicorn/server.py", line 75, in run
[WebServer]     return asyncio_run(self.serve(sockets=sockets), loop_factory=self.config.get_loop_factory())
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/asyncio/runners.py", line 195, in run
[WebServer]     return runner.run(main)
[WebServer]            ~~~~~~~~~~^^^^^^
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/asyncio/runners.py", line 118, in run
[WebServer]     return self._loop.run_until_complete(task)
[WebServer]            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
[WebServer]   File "uvloop/loop.pyx", line 1518, in uvloop.loop.Loop.run_until_complete
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/uvicorn/server.py", line 79, in serve
[WebServer]     await self._serve(sockets)
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/uvicorn/server.py", line 86, in _serve
[WebServer]     config.load()
[WebServer]     ~~~~~~~~~~~^^
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/uvicorn/config.py", line 441, in load
[WebServer]     self.loaded_app = import_from_string(self.app)
[WebServer]                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/site-packages/uvicorn/importer.py", line 19, in import_from_string
[WebServer]     module = importlib.import_module(module_str)
[WebServer]   File "/opt/hostedtoolcache/Python/3.13.12/x64/lib/python3.13/importlib/__init__.py", line 88, in import_module
[WebServer]     return _bootstrap._gcd_import(name[level:], package, level)
[WebServer]            ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[WebServer]   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
[WebServer]   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
[WebServer]   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
[WebServer]   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
[WebServer]   File "<frozen importlib._bootstrap_external>", line 1023, in exec_module
[WebServer]   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
[WebServer]   File "/home/runner/work/wordpack-for-english/wordpack-for-english/apps/backend/backend/main.py", line 386, in <module>
[WebServer]     app = create_app()
[WebServer]   File "/home/runner/work/wordpack-for-english/wordpack-for-english/apps/backend/backend/main.py", line 380, in create_app
[WebServer]     app.add_event_handler("shutdown", _on_shutdown)
[WebServer]     ^^^^^^^^^^^^^^^^^^^^^
[WebServer] AttributeError: 'FastAPI' object has no attribute 'add_event_handler'. Did you mean: 'add_exception_handler'?
Error: Process from config.webServer was not able to start. Exit code: 1

--- a/apps/backend/backend/main.py
+++ b/apps/backend/backend/main.py
@@ -377,8 +377,15 @@ def create_app() -> FastAPI:
     app.include_router(cfg.router, prefix="/api")
     app.include_router(tts.router, dependencies=protected_dependency)
 
-    app.add_event_handler("shutdown", _on_shutdown)
-    app.add_event_handler("startup", _on_startup_seed)
+    add_event_handler = getattr(app, "add_event_handler", None)
+    if callable(add_event_handler):
+        add_event_handler("shutdown", _on_shutdown)
+        add_event_handler("startup", _on_startup_seed)
+    else:
+        # FastAPI の新しいバージョンでは `FastAPI.add_event_handler` がなく、
+        # router 側の API を利用する必要がある。
+        app.router.add_event_handler("shutdown", _on_shutdown)
+        app.router.add_event_handler("startup", _on_startup_seed)
 
     return app
 

